### PR TITLE
[WIP] Add Mac OS X platform and compilers

### PR DESCRIPTION
**Pull request by ChrisNonyminus**

### CODE REVIEW

**Comment by ChrisNonyminus** [comment]
https://commons.wikimedia.org/wiki/File:Tiger-like-x.svg
This is the logo's page. It's in the public domain.

**Comment by ChrisNonyminus** [comment]
I should probably add it to the credits page though, yeah.

**Comment by ChrisNonyminus** [comment]
image
Right, sorry, I didn't realize you asked for a screenshot of the decomp.me site. I thought you meant to link to the source of the svg. I'm dumb :P

**Comment by ethteck** [comment]
np :). and hmmm I wonder why we're getting 3, 4, 2 platforms per row. without dummy I wonder what it'll look like..

**Comment by ChrisNonyminus** [comment]
Without dummy it looks like this.
image

**Comment by ethteck** [comment]
I'd also add a unit test that confirms a compilation works (like you did for the os 9 PR) if you don't mind

File: backend/compilers/download.py [hunk 1]
```diff
@@ -161,6 +161,82 @@ def download_zip(
...
+        dest_name="gcc-5363-cpp",
+    )
+    download_tar(
+        url="https://github.com/ChrisNonyminus/powerpc-darwin-cross/releases/download/initial/gcc3-1041.tar.gz",
+        dl_name="gcc3-1041.tar.gz",
+        dest_name="gcc3-1041",
+    )
+    download_file(
+        url="https://gist.githubusercontent.com/ChrisNonyminus/ec53837b151a65e4233fa53604de4549/raw/1633f3fac3d0075051105b9b4d66891086ec27bf/convert_gas_syntax.py",
+        log_name="convert_gas_syntax.py",
```
Comment (ethteck) [inline backend/compilers/download.py:205]: I don't think we should be re-downloading this file a bunch of times. We should be able to download it once and then copy it

**Review by ethteck — CHANGES_REQUESTED** [review]

**Pull request merged**